### ISSUE TITLE
Reusable workflow secrets/inputs/outputs should be case-insensitive

### ISSUE THREAD

**Issue by jorgellanos91** [rhysd/actionlint#216]
When I call a reusable workflow with a secret in the new version of the action (v1.6.18) it fails with 

` secret is not defined in reusable workflows` error.

```
.github/workflows/job.yml:17:11: secret “SECRET" is required by "./.github/workflows/reusable-job.yml" reusable workflow [workflow-call]
   |
17 |     uses: ./.github/workflows/reusable-job.yml
   |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.github/workflows/jobyml:21:7: secret “secret" is not defined in "./.github/workflows/reusable-job.yml" reusable workflow. defined secret is “SECRET" [workflow-call]
```
But when I call it I pass the secrets and still fail.

If I force the workflow to use the previous version (v1.6.17) without any other change then it succeeds.

Maybe it's related to the relative path of the reusable workflow.

**Comment by selfagency** [comment]
i'm having this issue as well with the latest version across all my workflows

**Comment by mariusfilipowski** [comment]
Same for me. It looks like the name of the secrets is expected in lower case although they are (as defined) in upper case.

**Comment by rhysd** [comment]
I could reproduce this:

reusable.yaml:

```yaml
on:
  workflow_call:
    secrets:
      SECRET:
        required: true

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - run: ...
```

test.yaml:

```yaml
on: push

jobs:
  test:
    uses: ./reusable.yaml
    secrets:
      secret: test
```

Errors reported:

```
test.yaml:5:11: secret "SECRET" is required by "./reusable.yaml" reusable workflow [workflow-call]
  |
5 |     uses: ./reusable.yaml
  |           ^~~~~~~~~~~~~~~
test.yaml:7:7: secret "secret" is not defined in "./reusable.yaml" reusable workflow. defined secret is "SECRET" [workflow-call]
  |
7 |       secret: test
  |       ^~~~~~~
```

**Comment by rhysd** [comment]
I think `inputs` also has the same issue though it is smaller issue since inputs are usually in lower case.

**Comment by rhysd** [comment]
I fixed this issue at 6713f82 but this issue will be fixed when more regression tests are added and the branch is merged into main.

**Comment by rhysd** [comment]
[v1.6.19](https://github.com/rhysd/actionlint/releases/tag/v1.6.19) was released.
